polyfill/test/zoneddatetime.mjs

```diff
@@ -256,6 +291,370 @@ describe('ZonedDateTime', () => {
+        equal(zdt.toString(), '2020-11-01T01:30:00-08:00[America/Los_Angeles]');
+      });
+      it("{ offset: 'prefer' } if offset does not match time zone", () => {
+        const zdt = ZonedDateTime.from('2020-11-01T04:00-07:00[America/Los_Angeles]', { offset: 'prefer' });
+        equal(zdt.toString(), '2020-11-01T04:00:00-08:00[America/Los_Angeles]');
+      });
+      it("{ offset: 'ignore' } uses time zone only", () => {
+        const zdt = ZonedDateTime.from('2020-11-01T04:00-12:00[America/Los_Angeles]', { offset: 'ignore' });
+        equal(zdt.toString(), '2020-11-01T04:00:00-08:00[America/Los_Angeles]');
+      });
+      it("{ offset: 'use' } uses offset only", () => {
+        const zdt = ZonedDateTime.from('2020-11-01T04:00-07:00[America/Los_Angeles]', { offset: 'use' });
+        equal(zdt.toString(), '2020-11-01T03:00:00-08:00[America/Los_Angeles]');
+      });
+      it('throw when bad offset', () => {
+        ['', 'PREFER', 'balance', 3, null].forEach((offset) => {
+          throws(() => ZonedDateTime.from('2020-11-01T04:00-07:00[America/Los_Angeles]', { offset }), RangeError);
+        });
+      });
+    });
+    describe('Disambiguation option', () => {
+      it('plain datetime with multiple instants - Fall DST in Brazil', () => {
+        const str = '2019-02-16T23:45[America/Sao_Paulo]';
+        equal(`${ZonedDateTime.from(str)}`, '2019-02-16T23:45:00-02:00[America/Sao_Paulo]');
+        equal(
+          `${ZonedDateTime.from(str, { disambiguation: 'compatible' })}`,
+          '2019-02-16T23:45:00-02:00[America/Sao_Paulo]'
+        );
+        equal(
+          `${ZonedDateTime.from(str, { disambiguation: 'earlier' })}`,
+          '2019-02-16T23:45:00-02:00[America/Sao_Paulo]'
+        );
+        equal(
+          `${ZonedDateTime.from(str, { disambiguation: 'later' })}`,
+          '2019-02-16T23:45:00-03:00[America/Sao_Paulo]'
+        );
+        throws(() => ZonedDateTime.from(str, { disambiguation: 'reject' }), RangeError);
+      });
+      it('plain datetime with multiple instants - Spring DST in Los Angeles', () => {
+        const str = '2020-03-08T02:30[America/Los_Angeles]';
+        equal(`${ZonedDateTime.from(str)}`, '2020-03-08T03:30:00-07:00[America/Los_Angeles]');
+        equal(
+          `${ZonedDateTime.from(str, { disambiguation: 'compatible' })}`,
+          '2020-03-08T03:30:00-07:00[America/Los_Angeles]'
+        );
+        equal(
+          `${ZonedDateTime.from(str, { disambiguation: 'earlier' })}`,
+          '2020-03-08T01:30:00-08:00[America/Los_Angeles]'
+        );
+        equal(
+          `${ZonedDateTime.from(str, { disambiguation: 'later' })}`,
+          '2020-03-08T03:30:00-07:00[America/Los_Angeles]'
+        );
+        throws(() => ZonedDateTime.from(str, { disambiguation: 'reject' }), RangeError);
+      });
+      it('uses disambiguation if offset is ignored', () => {
```

Comment:
The same tests apply with `prefer` if the offset is bogus. Should probably test that case.

Reply:
Good point. To be clear, the case is e.g. `Temporal.ZonedDateTime.from('2020-03-08T02:30-23:59[America/Los_Angeles]', { offset: 'prefer', disambiguation })`? This is what I've added tests for.